src/App.tsx

```diff
@@ -1,111 +1,321 @@
+          <div className="col-span-10 col-start-1">
+            <Tooltip
+              title={<Typography variant="body1">{datasetName}</Typography>}
+              placement="top"
+              TransitionComponent={Zoom}
+              TransitionProps={{ timeout: 500 }}
+              enterDelay={500}
+            >
+              <Typography variant="h5" className="dataset-name">
                 {datasetName}
               </Typography>
-            } 
-            placement='top' 
-            TransitionComponent={Zoom} 
-            TransitionProps={{ timeout: 500 }}
-            enterDelay={500}
-            >
-              <Typography variant="h5" className='dataset-name'>{datasetName}</Typography>
             </Tooltip>
             <Typography variant="subtitle1">from {nodeName}</Typography>
-            <Typography variant='subtitle2'>{numMatchingSubjects} subjects match / {datasetTotalSubjects} total subjects</Typography>
-          </Grid>
-          </Grid>
-          <Grid item>
+            <Typography variant="subtitle2">
+              {numMatchingSubjects} subjects match / {datasetTotalSubjects} total subjects
+            </Typography>
+          </div>
+          <div className="col-span-2 justify-self-end">
             <ButtonGroup>
               {imageModals.sort().map((modal) => (
                 <Button key={modal} variant="text" className={modalities[modal].style}>
                   {modalities[modal].name}
                 </Button>
               ))}
             </ButtonGroup>
-          </Grid>
-        </Grid>
+          </div>
+        </div>
       </CardContent>
     </Card>
-  )
+  );
 }
 
-function ResultContainer({result} : {result: Result[]}) {
+function ResultContainer({ result }: { result: Result[] }) {
   return (
-    <Stack spacing={2}>
-          {exampleResult.map((item) =>
-          <ResultCard key={item.dataset_uuid} nodeName={item.node_name} datasetName={item.dataset_name} datasetTotalSubjects={item.dataset_total_subjects} numMatchingSubjects={item.num_matching_subjects} imageModals={item.image_modals} />
-          )}
-    </Stack>
-  )
+    <div className="grid gap-4">
+      {result.map((item) => (
+        <ResultCard
+          key={item.dataset_uuid}
+          nodeName={item.node_name}
+          datasetName={item.dataset_name}
+          datasetTotalSubjects={item.dataset_total_subjects}
+          numMatchingSubjects={item.num_matching_subjects}
+          imageModals={item.image_modals}
+        />
+      ))}
+    </div>
+  );
 }
 
-function App() {
-  const [result, setResult] = useState<Result[]>([])
+function QueryForm({
+  diagnosisOptions,
+  assessmentOptions,
+  apiQueryURL,
+  onSubmitQuery,
+}: {
+  diagnosisOptions: APIOption[];
+  assessmentOptions: APIOption[];
+  apiQueryURL: string;
+  onSubmitQuery: (url: string) => void;
+}) {
+  const [minAge, setMinAge] = useState<string | null>(null);
+  const [maxAge, setMaxAge] = useState<string | null>(null);
```

Comment:
I think eventually we should probably treat these as numeric. OK to keep as strings for now, but I could imagine this getting confusing if we pass strings to a component that then needs to ensure that the "string" is not smaller than 0 and so on.

Maybe a TODO comment for now should make sure we remember to fix this in the final refactor

Reply:
This may be an issue cause the `TextField` itself returns a string even for type number but we can see if its possible to change that. I'll add a TODO.